src/asset.rs

```diff
@@ -591,10 +615,50 @@ impl Asset {
             mothballed_year, ..
         } = &self.state
         else {
-            panic!("Cannot mothballed an asset that hasn't been commissioned")
+            panic!("Cannot mothball an asset that hasn't been commissioned")
         };
         *mothballed_year
     }
+
+    /// Checks if the assets corresponds to a process that has a `unit_size` and is therefore divisible.
+    pub fn is_divisible(&self) -> bool {
+        self.process.unit_size.is_some()
+    }
+
+    /// Divides an asset if it is divisible and returns a vector of children
+    ///
+    /// The children assets are identical to the parent (including state) but with a capacity defined
+    /// by the `unit_size`. Only Future or Selected assets can be divided.
+    pub fn divide_asset(&self) -> Vec<AssetRef> {
+        assert!(
+            matches!(
+                self.state,
+                AssetState::Future { .. } | AssetState::Selected { .. }
+            ),
+            "Assets with state {0} cannot be divided. Only Future or Selected assets can be divided",
+            self.state
+        );
+        assert!(
+            self.is_divisible(),
+            "Only assets corresponding to processes with a unit size defined can be divided"
+        );
+
+        // Time to divide the asset until its capacity is all down to zero
+        let mut capacity = self.capacity;
+        let mut children = Vec::new();
+        while capacity > Capacity(0.0) {
+            let mut child = self.clone();
+            child.capacity = self
+                .process
+                .unit_size
+                .filter(|size| *size <= capacity)
+                .unwrap_or(capacity);
+            capacity -= child.capacity;
```

Comment:
Actually, given that `unit_size` _must_ be `Some` here, I think it would be clearer if we unwrap it above and do the check at the same time. Otherwise someone reading this might think we have to handle the case where we're dividing an asset with a `unit_size` of `None`, when that doesn't make sense.

So the check above could be:

```rs
let unit_size = self.process.unit_size.expect("Only assets corresponding to processes with a unit size defined can be divided");
```

Then this bit could just be:

```suggestion
            capacity -= unit_size.min(capacity);
```